yaml.go

```diff
@@ -285,18 +320,29 @@ func (n *yamlNode) Children() []*yamlNode {
 	return nodes
 }
 
-func (n *yamlNode) applyOnAllNodes(expand func(string) string) {
-	if n == nil {
-		return
-	}
+func (n *yamlNode) applyOnAllNodes(expand func(string) string) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			if _, ok := r.(runtime.Error); ok {
```

Comment:
yes, runtime.Error is returned for nil pointer exceptions/index out of bounds. Config internal errors are plain builtin errors.